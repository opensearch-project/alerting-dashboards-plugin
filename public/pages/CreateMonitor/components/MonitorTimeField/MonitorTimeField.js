/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { hasError, isInvalid } from '../../../../utils/validate';
import { validateTimeField } from './utils/validation';
import { FormikComboBox } from '../../../../components/FormControls';

const MonitorTimeField = ({ dataTypes }) => {
  // Default empty option + options from index mappings mapped to ui select form
  const dateFields = Array.from(dataTypes.date || []);
  const options = [].concat(dateFields).map((option) => ({ label: option }));
  return (
    <FormikComboBox
      name="timeField"
      formRow
      fieldProps={{ validate: validateTimeField(dateFields) }}
      rowProps={{
        label: 'Time field',
        helpText: 'Choose the time field you want to use for your x-axis',
        isInvalid,
        error: hasError,
        style: { paddingLeft: '10px' },
      }}
      inputProps={{
        placeholder: 'Select a time field',
        options,
        onChange: (options, field, form) => {
          form.setFieldValue('timeField', options[0].label);
        },
        isClearable: false,
        singleSelection: { asPlainText: true },
        'data-test-subj': 'timeFieldComboBox',
      }}
    />
  );
};

export default MonitorTimeField;

MonitorTimeField.propTypes = {
  dataTypes: PropTypes.object.isRequired,
};
