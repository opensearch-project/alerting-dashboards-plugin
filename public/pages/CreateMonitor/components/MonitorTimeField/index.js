/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import MonitorTimeField from './MonitorTimeField';

export default MonitorTimeField;
