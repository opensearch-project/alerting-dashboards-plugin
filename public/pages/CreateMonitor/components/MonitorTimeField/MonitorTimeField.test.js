/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { Formik } from 'formik';

import MonitorTimeField from './MonitorTimeField';

describe('MonitorTimeField', () => {
  test('renders', () => {
    const component = (
      <Formik initialValues={{}} onSubmit={() => {}}>
        <MonitorTimeField dataTypes={{}} />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });

  test.skip('displays no options', () => {
    const wrapper = mount(
      <Formik initialValues={{}} onSubmit={() => {}}>
        <MonitorTimeField dataTypes={{}} />
      </Formik>
    );
    // Default blank option
    expect(wrapper.find('select').props().children[1].length).toBe(1);
  });

  test.skip('displays options', () => {
    const wrapper = mount(
      <Formik initialValues={{}} onSubmit={() => {}}>
        <MonitorTimeField dataTypes={{ date: ['date1', 'date2', 'date3'] }} />
      </Formik>
    );
    // 3 options + default blank option
    expect(wrapper.find('select').props().children[1].length).toBe(4);
  });
});
