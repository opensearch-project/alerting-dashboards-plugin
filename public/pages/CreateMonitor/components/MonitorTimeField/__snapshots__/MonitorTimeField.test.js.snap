// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorTimeField renders 1`] = `
<div
  class="euiFormRow"
  id="timeField-form-row-row"
  style="padding-left:10px"
>
  <div
    class="euiFormRow__labelWrapper"
  >
    <label
      aria-invalid="false"
      class="euiFormLabel euiFormRow__label"
      for="timeField-form-row"
    >
      Time field
    </label>
  </div>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="euiComboBox"
      data-test-subj="timeFieldComboBox"
      name="timeField"
      role="combobox"
    >
      <div
        class="euiFormControlLayout"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <div
            class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
            data-test-subj="comboBoxInput"
            tabindex="-1"
          >
            <p
              class="euiComboBoxPlaceholder"
            >
              Select a time field
            </p>
            <div
              class="euiComboBox__input"
              style="font-size:14px;display:inline-block"
            >
              <input
                aria-controls=""
                data-test-subj="comboBoxSearchInput"
                id="timeField"
                role="textbox"
                style="box-sizing:content-box;width:1px"
                value=""
              />
              <div
                style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
              />
            </div>
          </div>
          <div
            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <button
              aria-label="Open list of options"
              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
              data-test-subj="comboBoxToggleListButton"
              type="button"
            >
              <div>
                EuiIconMock
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFormHelpText euiFormRow__text"
      id="timeField-form-row-help-0"
    >
      Choose the time field you want to use for your x-axis
    </div>
  </div>
</div>
`;
